Model.py

```diff
@@ -0,0 +1,429 @@
+Model inspired from:
+`"Transformer-based Spatial-Temporal Feature Learning for EEG Decoding " <https://arxiv.org/pdf/2106.11170.pdf>`.
+
+Usage: type "from Model import <class>" to use one of its classes.
+       type "from Model import <function>" to use one of its functions.
+
+Contributors: Ambroise Odonnat.
+"""
+
+import torch
+
+import torch.nn.functional as F
+import numpy as np
+
+from torch import nn
+from torch import Tensor
+from torch.autograd import Variable
+
+from einops import rearrange
+from einops.layers.torch import Rearrange, Reduce
+
+
+
+""" ********** Residual connection for better training ********** """
+
+class ResidualAdd(nn.Module):
+    
+    def __init__(self, fn):
+        
+        """    
+        Args:
+            fn: Sequence of layers.
+        """
+        
+        super().__init__()
+        self.fn = fn
+
+    def forward(self, x, **kwargs):
+        res = x
+        x = self.fn(x, **kwargs)
+        x += res
+        return x
+
+
+    
+""" ********** Spatial transforming ********** """
+
+class channel_attention(nn.Module):
+    
+    def __init__(self, linear_size, vector_size, dropout, negative_slope, kernel_size, stride):
+        
+        """    
+        Args:
+            linear_size (int): Size of input and output in linear layer,
+            scaling (int): Scaling in scaled dot-product attention ,
+            dropout (float): Value of dropout,
+            negative_slope (float): Slope in Leaky Relu,
+            kernel_size (int): Kernel size in average pooling layer,
+            stride (int): Value of stride in average pooling layer.
+        """
+        
+        super().__init__()
+
+        # Query layer
+        self.query = nn.Sequential(
+            nn.Linear(linear_size, linear_size),
+            nn.LayerNorm(linear_size),  # also may introduce improvement to a certain extent
+            nn.Dropout(dropout)
+        )
+        
+        # Key layer
+        self.key = nn.Sequential(
+            nn.Linear(linear_size, linear_size),
+            # nn.LeakyReLU(),
+            nn.LayerNorm(linear_size),
+            nn.Dropout(dropout)
+        )
+
+        # Final projection layer
+        self.projection = nn.Sequential(
+            nn.Linear(linear_size, linear_size),
+            nn.LeakyReLU(negative_slope),
+            nn.LayerNorm(linear_size),
+            nn.Dropout(dropout)
+        )
+
+        self.dropout = nn.Dropout(0)
+        
+        # Average pooling layer
+        self.pooling = nn.AvgPool2d(kernel_size=(1, kernel_size), stride=(1, stride)) 
+
+        # Compute Query dimension after average pooling
+        query_size = vector_size - kernel_size
+        query_size /= stride
+        query_size += 1
+        self.scaling = query_size ** (1 / 2) 
+        
+        # Weights initiation
+        for m in self.modules():
+            if isinstance(m, nn.Linear):
+                nn.init.xavier_normal_(m.weight, gain = 1)
```

Comment:
Justification for xavier normal?

Reply:
It was the one in the original model and I tested all the other methods proposed by torch: it seems to be the best one